transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -311,19 +317,24 @@ private void invokeChannelRead(Object msg) {
         }
     }
 
+    private void invokeNextChannelRead(AbstractChannelHandlerContext next, Object msg) {
+        nextChannelReadInvoked = true;
```

Comment:
Shoildn't this method be executed only by the correct Thread? What I'm missing?



> Am 07.02.2015 um 06:05 schrieb Trustin Lee <notifications@github.com>:
> 
> In transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java:
> 
> > @@ -311,19 +317,24 @@ private void invokeChannelRead(Object msg) {
> >          }
> >      }
> >  
> > +    private void invokeNextChannelRead(AbstractChannelHandlerContext next, Object msg) {
> > +        nextChannelReadInvoked = true;
> If I understood correctly, nextChannelReadInvoked can be updated by any non-I/O thread, because this method is being executed by the potentially different executor than the executor of the current context.
> 
> —
> Reply to this email directly or view it on GitHub.
>